invokeai/app/services/download/download_default.py

```diff
@@ -188,6 +190,16 @@ def cancel_all_jobs(self) -> None:
             if not job.in_terminal_state:
                 self.cancel_job(job)
 
+    def wait_for_job(self, job: DownloadJob, timeout: int = 0) -> DownloadJob:
+        """Block until the indicated job has reached terminal state, or when timeout limit reached."""
+        start = time.time()
+        while not job.in_terminal_state:
+            if self._job_completed_event.wait(timeout=5):  # in case we miss an event
```

Comment:
Agree. I don't think the right solution is to use the caller's timeout, though, because what we're waiting on is one of possibly multiple events and we're basically polling and I don't think a spin-loop would be good here. I've hard-coded the inner timeout to 0.25s, which means we could overshoot by a quarter of a second. Is this an OK compromise?